lib/segment/src/segment_constructor/segment_constructor_base.rs

```diff
@@ -942,3 +948,40 @@ fn load_segment_state_v5(segment_path: &Path) -> OperationResult<SegmentState> {
             ))
         })
 }
+
+/// Migrate a RocksDB based ID tracker into a mutable ID tracker.
+///
+/// Creates a new mutable ID tracker and copies all mappings from the RocksDB based ID tracker into
+/// it. The persisted RocksDB data is deleted so that only the new tracker will be loaded next
+/// time. The new ID tracker is returned.
+#[cfg(feature = "rocksdb")]
+pub fn migrate_rocksdb_id_tracker_to_mutable(
+    old_id_tracker: SimpleIdTracker,
+    segment_path: &Path,
+) -> OperationResult<MutableIdTracker> {
+    log::info!("Migrating ID tracker from RocksDB into new format");
+
+    // Construct mutable ID tracker
+    let mut new_id_tracker = create_mutable_id_tracker(segment_path)?;
+    assert_eq!(
+        new_id_tracker.total_point_count(),
+        0,
+        "new mutable ID tracker must be empty",
+    );
+
+    // Copy all mappings into it
+    for (external_id, internal_id) in old_id_tracker.iter_from(None) {
+        let version = old_id_tracker.internal_version(internal_id).unwrap_or(0);
```

Comment:
Why are we creating a synthetic version here?
If the version is missing it means the point is not well formed I believe.